### ISSUE TITLE
Changing sku azurerm_servicebus_namespace from Basic to Standard (and the other way around) should not cause recreation.

### ISSUE THREAD

**Issue by w0ut0** [hashicorp/terraform-provider-azurerm#16463]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.0.9

### AzureRM Provider Version

2.99.0

### Affected Resource(s)/Data Source(s)

azurerm_servicebus_namespace

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "terraform-servicebus"
  location = "West Europe"
}

resource "azurerm_servicebus_namespace" "example" {
  name                = "tfex-servicebus-namespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"

  tags = {
    source = "terraform"
  }
}
```

### Debug Output/Panic Output

```shell
# module.servicebus.azurerm_servicebus_namespace.sb must be replaced
-/+ resource "azurerm_servicebus_namespace" "example" {
      ~ default_primary_connection_string   = (sensitive value)
      ~ default_primary_key                 = (sensitive value)
      ~ default_secondary_connection_string = (sensitive value)
      ~ default_secondary_key               = (sensitive value)
      ~ id                                  = "/subscriptions/xxxx/resourceGroups/xxxx/providers/Microsoft.ServiceBus/namespaces/tfex-servicebus-namespace" -> (known after apply)
        name                                = "devtfex-servicebus-namespace"
      ~ sku                                 = "Basic" -> "Standard" # forces replacement
      - tags                                = {} -> null
      - zone_redundant                      = false -> null
        # (3 unchanged attributes hidden)
    }
```

### Expected Behaviour

Switching between `Basic` and `Standard` should not require a recreation (therefor losing all queues, messages,...). Confirmed to work in the portal.

### Actual Behaviour

Terraform will recreate the servicebus, therefor losing all (externally created) queues and messages.

### Steps to Reproduce

`terraform apply`

A workaround is changing the SKU in the portal before running `terrafrom apply`.

**Comment by xiaxyi** [comment]
@w0ut0  Checked the api behavior and find out the sku can be updated from basic to standard. Need to have a future check from TF side. will update accordingly

**Comment by jackofallops** [comment]
Closed by #16523.
